parsers/IL.py

```diff
@@ -18,29 +18,46 @@
 )
 PRICE = "50.66"  # Price is determined yearly
 
+def flatten_list(_2d_list):
+    flat_list = []
+    # Iterate through the outer list
+    for element in _2d_list:
+        if type(element) is list:
+            # If the element is of type list, iterate through the sublist
+            for item in element:
+                flat_list.append(item)
+        else:
+            flat_list.append(element)
+    return flat_list
 
 def fetch_production(zone_key="IL", session=None, logger=None):
     first = requests.get(IEC_PRODUCTION)
     first.cookies
-    second = requests.get(IEC_PRODUCTION, cookies=first.cookies)
+    with requests.get(IEC_PRODUCTION, cookies=first.cookies) as second:
```

Comment:
Does using a long context manager here add something? Also immediately above there's a statement with no effect.

Reply:
This is part of IL parser which do not belong here. will rebase